main.py

```
@@ -1,279 +1,144 @@
+# --- Bot Configuration & Credentials ---
+# Load essential credentials and tokens from environment variables.
 SENDER_EMAIL = os.getenv("SENDER_EMAIL")
 SENDER_PASSWORD = os.getenv("SENDER_PASSWORD")
 BOT_TOKEN = os.getenv("DISCORD_BOT_TOKEN")
 
+# Critical check: Ensure the bot token is available. Exit if not.
+if not BOT_TOKEN:
+    print("CRITICAL ERROR: DISCORD_BOT_TOKEN not found in environment variables.")
+    print("Please ensure a .env file is present in the root directory with DISCORD_BOT_TOKEN set,")
+    print("or that the environment variable is set in the deployment environment.")
+    exit() # Stop execution if token is missing
```

Pylint (3.3.7)

[refactor] Consider using 'sys.exit' instead

(R1722)